### PR TITLE
Separate plugin functionality and do some code cleanup

### LINKED ISSUE #4: Separate plugin functionality
I want to separate the functionality of the plugin into the side that people will use to record their gameplay as training data for Neuro and the side that communicates with the AI to allow Neuro to play.

I've started by moving some of the Neuro-specific functionality into the Vision class (this keeps track of what Neuro sees throughout the round to be fed into the language model). We should probably use the Recorder class for gathering Frame data.

We will also probably want a separate class in the future to interface with the Python neural network, probably just open a socket and have some sort of communication protocol. This may end up being a different plugin(?), that instead just uses the data collected from this plugin to interface with the neural network.

Ideally there shouldn't be much code in the NeuroPlugin class too and it should instead handle setting these individual components up.

### PR DESCRIPTION
**Pull request by AlexejheroYTB**
Closes #4

I am opening this draft PR in advance to receive individual feedback for separate commits as I am working on restructuring the project. 

The larger commits include a description of the individual changes that were made.

The plan for separating functionality is to use MonoBehaviours and dependency injection.

---

### Changes made in this pull request

- Updated dependencies to the latest version
- Implemented dependency injection
- Removed basically everything from NeuroPlugin.cs
- ... (will add later)

### CODE REVIEW

**Comment by AlexejheroYTB** [comment]
When I implement the vision handler I will try to figure out what the numbers are for and turn them into constants if that is necessary

File: Neuro/zzz/OldVision.cs [hunk 1]
```diff
@@ -102,27 +101,27 @@ public void UpdateVision()
...
                     {
                         if (PlayerControl.LocalPlayer == potentialWitness) continue;
 
                         if (potentialWitness.inVent || potentialWitness.Data.IsDead) continue;
 
-                        if (Vector2.Distance(potentialWitness.transform.position, deadBody.transform.position) < 3f)
-                        {
-                            witnesses.Add(potentialWitness);
-                        }
+                        if (Vector2.Distance(potentialWitness.transform.position, deadBody.transform.position) < 3f) witnesses.Add(potentialWitness);
```
[the remark this answers is not in the document]
Comment (AlexejheroYTB) [reply Neuro/zzz/OldVision.cs]: Resolving these as the vision handler needs to be rewritten entirely anyway

**Comment by JohnyDaison** [comment]
So you don't plan to merge this PR at all? There will be smaller PRs with parts of these changes? Because this is getting too big to properly review.

**Comment by AlexejheroYTB** [comment]
> So you don't plan to merge this PR at all? There will be smaller PRs with parts of these changes? Because this is getting too big to properly review.

No, I plan to merge this PR obviously. Once I test it then it will be ready for merge, but I will want for #7 and #8 to be merged before so I can handle merge conflicts. As for the review, you can review this PR on a commit-by-commit basis.

**Comment by Vedal987** [comment]
Apologies for the force push, feel free to reopen or create a new PR with these changes.

**Pull request closed without merging**